app/presenters/curation_concerns/single_use_link_presenter.rb

```diff
@@ -0,0 +1,35 @@
+module CurationConcerns
+  class SingleUseLinkPresenter
+    include ActionView::Helpers::TranslationHelper
+
+    attr_reader :link, :link_type, :url_helper
+
+    delegate :downloadKey, :expired?, :to_param, to: :link
+
+    # @param link [SingleUseLink]
+    def initialize(link)
+      @link = link
+      define_link
+    end
+
+    def human_readable_expiration
+      hours = (link.expires - Time.zone.now).to_i / 3600
+      return t('curation_concerns.single_use_links.expiration.lesser_time') if hours < 1
+      t('curation_concerns.single_use_links.expiration.time', value: hours)
+    end
+
+    def short_key
+      link.downloadKey.first(6)
+    end
+
+    def define_link
+      if link.path =~ /downloads/
+        @link_type = t('curation_concerns.single_use_links.download.type')
```

Comment:
No performance impact that I'm aware of. It's more in line with the lessons we learned about OOP design from Sandi Metz, namely that it's preferable to send a message than to rely on an ivar, and it's also easier to stub and override methods than it is to deal with ivars. It's also one of the "Sandi rules" to make it so that a controller only has to know about one ivar, which in our world is typically a presenter instance.